src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs

```diff
@@ -20,6 +20,8 @@ public static class ValidationUtilities
         private static readonly Regex ResourceProviderPathPattern = new Regex(@"/providers/(?<resPath>[^{/]+)/", RegexOptions.IgnoreCase);
         private static readonly Regex PropNameRegEx = new Regex(@"^[a-z0-9\$-]+([A-Z]{1,3}[a-z0-9\$-]+)+$|^[a-z0-9\$-]+$|^[a-z0-9\$-]+([A-Z]{1,3}[a-z0-9\$-]+)*[A-Z]{1,3}$");
 
+        private static readonly Regex ResourceProviderNamespaceRegex = new Regex(@"(resource-manager|data-plane)/(?<namespace>[\w\.]+)/");
```

Comment:
For this case, we are not updating an existing rule, actually, we are retiring R2059 and use a new rule instead.
What if we have a new rule id for this?

Reply:
Agree with Renhe, it seems to be a new rule, instead of upgrade the original rule R2025.  How about we set up a new rule?